docs/static/config-details.asciidoc

```diff
@@ -77,15 +87,74 @@ process.
 info, see <<profiling-the-heap>>.
 // end::heap-size-tips[]
 
-
 [[off-heap-size]]
-==== Setting the off-heap size
+===== Setting the off-heap size
 
 The operating system, persistent queue mmap pages, direct memory, and other processes require memory in addition to memory allocated to heap size.
-Keep the overall memory requirements in mind when you allocate memory.
 
+Internal JVM data structures, thread stacks, memory mapped files and direct memory for input/output (IO) operations are all parts of the off-heap JVM memory.
+Thread stacks zone contains the list of stack frames for each Java thread spinned by the JVM; each frame keeps the local arguments passed during method calls.
+Read on <<stacks-size>> if the size needs to be adapted to the processing needs.
+Memory mapped files are not part of the Logstash's process off-heap memory, but consume RAM when paging files from disk.
+These mapped files speed up the access to Persistent Queues pages, a performance improvement - or trade off - to reduce expensive disk operations such as read, write, and seek.
+Some network I/O operations also resort to in-process direct memory usage to avoid, for example, copying of buffers between network sockets,
+in particular, some input plugins such as Elastic Agent, Beats, TCP, and HTTP inputs, use direct memory.
+
+Another component of memory consumption is the thread stack size, allocated per thread.
+Plugins, depending on their type (inputs, filters, and outputs), have different thread models.
+Every input plugin runs in its own thread and can potentially spawn others. For example, each JDBC input
+plugin launches a scheduler thread. Netty based plugins like TCP, Beats or HTTP input spawn a thread pool with 2 * number_of_cores threads.
+Output plugins may also start helper threads, such as a connection management thread for each
+{es} output instance.
+Every pipeline, also, has its own thread responsible to manage the pipeline lifecycle.
+
+To summarize, we have 3 categories of memory usage, where 2 can be limited by the JVM and the other relies on available, free memory:
+
+| Memory Type | Configured using | Used by |
```

Comment:
```suggestion
[cols="<,<,<",options="header",]
|=====
| Memory Type | Configured using | Used by
| JVM Heap  |   -Xmx   | any normal object allocation
| JVM direct memory |   -XX:MaxDirectMemorySize   | beats, tcp and http inputs
| Native memory  |  N/A   | Persistent Queue Pages, Thread Stacks
|=====
```

Reply:
@andsel, this formatting should fix your table problem.  There are some other formatting and content issues that need to be fixed as well, but I know that you were particularly concerned about this table.